### ISSUE TITLE
What is the max number prompts that the generate() method can take

### ISSUE THREAD

**Issue by hxue3** [vllm-project/vllm#1315]
What is the max number prompts that the llm.generate() method can take?
I know vllm uses continus batching, but I want to dobule check if we can feed any num of prompts for this method

**Comment by WoosukKwon** [comment]
Hi @hxue3, thanks for the question. The `LLM` class automatically batches the input sequences based on the available memory in your GPU and returns the results after every input is processed. Therefore, you can put any number of prompts in `llm.generate()` and it's indeed the best way to maximize the throughput.

**Comment by beyondguo** [comment]
@WoosukKwon I noticed that, the more prompts I feed into the function, the lower process speed I got. For example, when I only input 1k prompts, the speed input/output speed is about 17000/5000 toks/s, when I input 10k prompts, the input/output speed is about 7000/2000 toks/s. 

I have about 1 million prompts to process. If I push all these prompts into the fucntion, it took quite a long time before the "Processed prompts" bar shows up. In this waiting period, the CPU utilization on my machine is rather low (nearly 0), so I'm wondering what is going on in this period? Is it the tokenization process for all the prompts? If so, the CPU utilization should be high.
